### PR TITLE
#100 use pipeline credentials depending on credName on BU

### PR DESCRIPTION
**Pull request by DannyCastroVentura**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

- [X] Bug fix
- [X] Enhanced Copado function

## What changes did you make? (Give an overview)

Changed the retrieve.js, commit.js and deploy.js code:
- working with the credentials of the pipeline field and environment credential name.

### CODE REVIEW

File: copado-function/app/Retrieve.js [hunk 1]
```diff
@@ -98,6 +85,12 @@ async function run() {
     Util.convertEnvVariables(CONFIG.envVariables);
     Log.debug(CONFIG);
 
+    // ensure we got SFMC credentials for our source BU
+    if (!CONFIG.credentials[CONFIG.credentialNameSource]) {
+        throw new Error(`No credentials found for source (${CONFIG.credentialNameSource})`);
```
Comment (JoernBerkefeld) [inline copado-function/app/Retrieve.js:90]:
```suggestion
        Log.error(`No credentials found for source (${CONFIG.credentialNameSource})`);
        throw new Error(`No credentials`);
```

File: copado-function/app/Commit.js [hunk 2]
```diff
@@ -35,22 +35,9 @@ const resolve = require('node:path').resolve;
...
-        target: {
-            clientId: process.env.clientId,
-            clientSecret: process.env.clientSecret,
-            credentialName: process.env.credentialName,
-            tenant: process.env.tenant,
-            enterpriseId: process.env.enterprise_id,
-        },
-    },
+    credentialNameSource: process.env.credentialNameSource,
+    credentialNameTarget: process.env.credentialNameTarget,
```
Comment (JoernBerkefeld) [inline copado-function/app/Commit.js:39]:
```suggestion
    credentialNameTarget: null,
```

File: copado-function/app/Deploy.js [hunk 3]
```diff
@@ -117,6 +104,16 @@ async function run() {
     Util.convertEnvVariables(CONFIG.envVariables);
     Log.debug(CONFIG);
 
+    // ensure we got SFMC credentials for our source BU
+    if (!CONFIG.credentials[CONFIG.credentialNameSource]) {
+        throw new Error(`No credentials found for source (${CONFIG.credentialNameSource})`);
```
Comment (JoernBerkefeld) [inline copado-function/app/Deploy.js:109]:
```suggestion
        Log.error(`No credentials found for source (${CONFIG.credentialNameSource})`);
        throw new Error(`No source credentials`);
```

File: copado-function/app/Deploy.js [hunk 4]
```diff
@@ -117,6 +104,16 @@ async function run() {
...
+    }
+
+    // ensure we got SFMC credentials for our target BU
+    if (!CONFIG.credentials[CONFIG.credentialNameTarget]) {
+        throw new Error(`No credentials found for target (${CONFIG.credentialNameTarget})`);
```
Comment (JoernBerkefeld) [inline copado-function/app/Deploy.js:114]:
```suggestion
        Log.error(`No credentials found for target (${CONFIG.credentialNameTarget})`);
        throw new Error(`No target credentials`);
```

File: copado-function/app/Deploy.js [hunk 5]
```diff
@@ -107,7 +94,7 @@ const CONFIG = {
 /**
  * main method that combines runs this function
  *
- * @returns {void}
+ * @returns {void}credSource
```
Comment (JoernBerkefeld) [inline copado-function/app/Deploy.js:97]:
```suggestion
 * @returns {void}
```
accidental change?

File: copado-function/app/Commit.js [hunk 6]
```diff
@@ -98,6 +85,11 @@ async function run() {
     Util.convertEnvVariables(CONFIG.envVariables);
     Log.debug(CONFIG);
 
+    // ensure we got SFMC credentials for our source BU
+    if (!CONFIG.credentials[CONFIG.credentialNameSource]) {
+        throw new Error(`No credentials found for source (${CONFIG.credentialNameSource})`);
```
Comment (JoernBerkefeld) [inline copado-function/app/Commit.js:90]:
```suggestion
        Log.error(`No credentials found for source (${CONFIG.credentialNameSource})`);
        throw new Error(`No source credentials`);
```

File: copado-function/app/Deploy.config [hunk 7]
```diff
@@ -14,14 +14,11 @@
...
-* [clientId] ({$Pipeline.Property.mc_client_id})
-* [clientSecret] ({$Pipeline.Property.mc_client_secret})
-* [tenant] ({$Pipeline.Property.mc_tenant})
 * [toBranch] ({$Context.JobExecution__r.DataJSON.destinationBranchName}) 
 * [promotionBranch] ({$Context.JobExecution__r.DataJSON.promotionBranchName})
 * [merge_strategy] (theirs)
-* [enterprise_id] ({$Pipeline.Property.enterprise_id})
 * [metadataFileId] ({$Context.JobExecution__r.DataJson.fileWithSelectedChanges})
 * [metadata_file] ({$Context.JobExecution__r.DataJson.fileWithSelectedChanges})
-* [main_branch] ({$Pipeline.copado__Main_Branch__c})
```
Comment (JoernBerkefeld) [inline copado-function/app/Deploy.config:27]: why was this removed?

**Review by JoernBerkefeld — CHANGES_REQUESTED** [review]

File: copado-function/app/Deploy.config (hunk 7 above)
Comment (DannyCastroVentura) [reply copado-function/app/Deploy.config:27]: mistake, already fixed!

File: copado-function/app/Deploy.js (hunk 5 above)
Comment (DannyCastroVentura) [reply copado-function/app/Deploy.js]: Yes

**Review by JoernBerkefeld — APPROVED** [review]
